### PR TITLE
Allow subparsers to have Usage strings

### PR DESCRIPTION
**Pull request by Juice805**
In larger command line tools there can be a few levels of subparsers and a top level USAGE description may not be able to capture all varieties of sub-commands. It would help users of these tools if they can get USAGE strings for lower level sub parsers.

### CODE REVIEW

**Comment by Juice805** [comment]
@swift-ci smoke test

**Comment by aciidb0mb3r** [comment]
Can you add a test?

**Comment by Juice805** [comment]
added assert `testSubparser` to test `USAGE` string

File: Sources/SPMUtility/ArgumentParser.swift [hunk 1]
```diff
@@ -889,7 +889,7 @@ public final class ArgumentParser {
         stream <<< "OVERVIEW: " <<< overview
 
         // We only print command usage for top level parsers.
```
Comment (aciidb0mb3r) [inline Sources/SPMUtility/ArgumentParser.swift]: Can you update/remove this comment?

Comment (Juice805) [reply Sources/SPMUtility/ArgumentParser.swift]: Done! Good catch

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Pull request merged**